src/domain/session/toast/ToastCollectionViewModel.ts

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import {CallToastNotificationViewModel} from "./CallToastNotificationViewModel";
+import {ObservableArray} from "../../../observable";
+import {ViewModel, Options as BaseOptions} from "../../ViewModel";
+import type {GroupCall} from "../../../matrix/calls/group/GroupCall";
+import type {Room} from "../../../matrix/room/Room.js";
+import type {Session} from "../../../matrix/Session.js";
+import type {SegmentType} from "../../navigation";
+
+type Options = {
+    session: Session;
+} & BaseOptions;
+
+export class ToastCollectionViewModel extends ViewModel<SegmentType, Options> {
+    public readonly toastViewModels: ObservableArray<CallToastNotificationViewModel> = new ObservableArray();
+
+    constructor(options: Options) {
+        super(options);
+        const session = this.getOption("session");
+        const callsObservableMap = session.callHandler.calls;
+        this.track(callsObservableMap.subscribe(this));
```

Comment:
Any reason you need to  manually handle the observablemap events below, rather than use `MappedMap`  and `SortedList` to get an obserable list of view models, e.g. use the `mapValues` and `sortValues` operators? E.g. something like this:
```js
const viewModelList = callHandler.calls.mapValues(
            (member, emitChange) => new CallToastNotificationViewModel(this.childOptions({ call, room, dismiss }))
        ).sortValues((aVM, bVM) => /* some sorting criteria, take care that the sorting criteria is stable, e.g. always returns the same order and that a and b can be the same value */);

Reply:
I don't see how we can delete a value from the `MappedMap` (it extends `BaseObservableMap` and not `ObservableMap`). I think the `MappedMap` works best in situations where a key in the source will always be in the  result as well i.e a consistent 1:1 mapping across the lifetime of the source.
Here, we need to remove the vm when the room is opened.

Also I feel like the use of a `SortedMapList` here would be a bit hacky since we really don't really need to sort for this to work.

I also feel like just having a `source -> subscriber` is more simple than having `source -> subscriber1 -> subscriber2 -> ...`, it's a bit more harder to think through especially in future when we'll have to support arbitrary notification sources.

Reply:
Right, that makes sense.